python/tchannel/tornado/stream.py

```diff
@@ -98,27 +103,34 @@ def __init__(self, buf=None, auto_close=True):
 
         :param buf: the buffer for the in memory stream
         """
-        self._stream = deque()
+        self.stream = deque()
         if buf:
-            self._stream.append(buf)
+            self.stream.append(buf)
         self.state = StreamState.init
         self._condition = Condition()
         self.auto_close = auto_close
 
         self.exception = None
 
+    def clone(self):
+        new_stream = self.__class__()
+        new_stream.state = self.state
+        new_stream.auto_close = self.auto_close
+        new_stream.stream = copy.deepcopy(self.stream)
```

Comment:
What's stopping you from simply doing `deque(self._stream)`? Deques are ordered iterables. Copies will be the same. They contain immutable strings. There's no reason to use `deepcopy`.